<!DOCTYPE html>
<html lang="ja">
    <head>
        <link rel="shortcut icon" href="https://iwasakishuto.github.io/images/apple-touch-icon/University.png" />

        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <meta name="author" content="Shuto" />

        <meta property="og:type" content="article" />
        <meta name="twitter:card" content="summary">

<meta name="keywords" content="3A, 情報基礎実験(浅井), 情報基礎実験(浅井), " />
<meta property="og:image" content="https://iwasakishuto.github.io/images/FacebookImage/University.png"/>

<meta property="og:title" content="RNA-seqデータ解析（バッチスクリプト版） "/>
<meta property="og:url" content="https://iwasakishuto.github.io/University/3A/情報基礎実験(浅井)-12.html" />
<meta property="og:description" content="" />
<meta property="og:site_name" content="3A" />
<meta property="og:article:author" content="Shuto" />
<meta property="og:article:published_time" content="2019-10-17T00:00:00+09:00" />
<meta property="og:article:modified_time" content="2019-10-17T00:00:00+09:00" />
<meta name="twitter:title" content="RNA-seqデータ解析（バッチスクリプト版） ">
<meta name="twitter:description" content="">

        <title>RNA-seqデータ解析（バッチスクリプト版）  · 3A
</title>
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/css/bootstrap-combined.v2.3.2.min.css">
        <link href="//netdna.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.css" rel="stylesheet">
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/University/3A/theme/css/pygments.css" media="screen">
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/University/3A/theme/tipuesearch/tipuesearch.css" media="screen">
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/University/3A/theme/css/elegant.css">
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/University/3A/theme/css/admonition.css" media="screen">
        <!---->
        <link rel="apple-touch-icon" sizes="152x152" href="https://iwasakishuto.github.io/images/apple-touch-icon/University.png" type="image/png" />
<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
     (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
     m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
     })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
    ga('create', 'UA-164333757-1', 'auto');
    ga('send', 'pageview');
</script>



        <!-- Use fontawesome Icon -->
        <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.0/css/all.css" integrity="sha384-lZN37f5QGtY3VHgisS14W3ExzMWZxybE1SJSEsQp9S+oqd12jhcu+A56Ebc1zFSJ" crossorigin="anonymous">
        <!-- Syntax highlight -->
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/8.6/styles/github.min.css">
        <!-- Custom CSS -->
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/css/custom.css">
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/8.6/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>
        <!-- LaTex -->
        <!-- Github env -->
        <!--<script type="text/javascript" async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML"></script>-->
        <script async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS_CHTML"></script>
        <script type="text/x-mathjax-config">
        	MathJax.Hub.Config({
        		tex2jax: {
        			inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        			displayMath: [ ['$$','$$'], ["\\[","\\]"] ]
        		}
        	});
        </script>
        <!-- Mermaid -->
        <script src="https://unpkg.com/mermaid/dist/mermaid.min.js" charset="UTF-8"></script>
        <script>
          mermaid.initialize({
            startOnLoad:true
          });
        </script>
        <script src="https://iwasakishuto.github.io/js/smooth-scroll.polyfills.min.js"></script>
    </head>
    <body>
        
        <div id="content-sans-footer">
        <div class="navbar navbar-static-top">
            <div class="navbar-inner">
                <div class="container-fluid">
                    <a class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </a>
                    <a class="brand" href="https://iwasakishuto.github.io/University/3A"><span class=site-name style="color: #80273F;"><i class="fa fa-book"></i> 3A</span></a>
                    <!--
                    <a class="brand" href="https://iwasakishuto.github.io/University/3A/"><span class=site-name style="margin-left:auto; margin-right:auto;"><i class="fas fa-book-reader"></i>3A</span></a>
                    -->
                    <div class="nav-collapse collapse">
                        <ul class="nav pull-right top-menu">
                            <li ><a href="https://iwasakishuto.github.io/"><i class="fas fa-home"></i>Portfolio Top</a></li>
                            <li ><a href="https://iwasakishuto.github.io/University/3A/categories"><i class="fa fa-list-alt"></i> Categories</a></li>
                            <li ><a href="https://iwasakishuto.github.io/University/3A/tags"><i class="fa fa-tags"></i> Tags</a></li>
                            <li ><a href="https://iwasakishuto.github.io/University/3A/archives"><i class="fa fa-folder-open"></i> Archives</a></li>
                            <li><form class="navbar-search" action="https://iwasakishuto.github.io/University/3A/search.html" onsubmit="return validateForm(this.elements['q'].value);"> <input type="text" class="search-query" placeholder="Search" name="q" id="tipue_search_input"></form></li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
        <div class="container-fluid">
            <div class="row-fluid">
                <div class="span1"></div>
                <div class="span10">
<article itemscope>
<div class="row-fluid">
  <header class="page-header span10 offset2">
    <h1><a href="https://iwasakishuto.github.io/University/3A/情報基礎実験(浅井)-12.html"> RNA-seqデータ解析（バッチスクリプト版）  </a></h1>
  </header>
</div>

<div class="row-fluid">
    <div class="span8 offset2 article-content">

  
  <p>今回は、<a href="./情報基礎実験(浅井)-11.html">RNA-seqデータ解析（対話モード版）</a>で行なった処理を毎回対話モードで行うのは面倒なので、バッチスクリプトを作成して一括で行います。</p>
<p>以下のパラメタファイル（<a href="theme/img/3A_Basic_Laboratory/RNAseq/params.json" download><code>params.json</code></a>）を読み込んで各種操作を行います。</p>
<p>```json
{
  "qsub_q": "u-debug",
  "n_node": 1,
  "n_cpus": 1,
  "n_MPIprocs": 1,
  "n_OMPthreads": 8,
  "walltime": "0:30:00",
  "group_list": "gj29",</p>
<p>"ID": "ERR315326",</p>
<p>"RNAseqData": "ftp://ftp.ddbj.nig.ac.jp/ddbj_database/dra/sralite/ByExp/litesra/ERX/ERX288/ERX288491/ERR315326/ERR315326.sra",
  "MappingRefGenome": "https://cloud.biohpc.swmed.edu/index.php/s/grch38/download",
  "GeneAnnotation": "ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.annotation.gtf.gz"
}
```</p>
<p>各変数は、 <a href="https://stedolan.github.io/jq/download/"><code>jq</code>コマンド</a>を使って以下のように読み込みます。</p>
<p><code>sh
[変数名]=$(cat [パラメタファイルへのパス] | jq -r .[key名])
[変数名]=`jq -r .[key名] [パラメタファイルへのパス]`</code></p>
<h2>成果物</h2>
<p><code>make_batch_u_qsub.sh</code> という以下のシェルスクリプトを叩くことで、<code>params.json</code> の内容を踏まえたバッチスクリプト <code>batch_u_qsub.sh</code> を作成します。</p>
<h3><a href="theme/img/3A_Basic_Laboratory/RNAseq/make_batch_u_qsub.sh"><code>make_batch_u_qsub.sh</code></a> <a href="" download>[download]</a></h3>
<p>```sh</p>
<h1>!/bin/bash</h1>
<p>Ssymbol="["
Esymbol="]"</p>
<p>function AssignParams () {
  &lt;&lt; COMMENTOUT
  @params params.json     {"qsub_q": "u-debug"}
  @params templates.sh    PBS -q [${n_node}]
  @return batch_u_qsub.sh PBS -q u-debug
  COMMENTOUT</p>
<p>line="<span class="math">\({1}"
  while true
  do
    Spos=<code>echo "${line}" | awk -v pattern=${Ssymbol} '{print index($0, pattern)}'</code>
    Epos=<code>echo "${line}" | awk -v pattern=${Esymbol} '{print index($0, pattern)}'</code>
    if [ ${Spos} -ne 0 -a ${Epos} -ne 0 -a ${Spos} -lt ${Epos} ]; then
      PREFIX=<code>echo "${line:0:$(($Spos-1))}"</code>
      VARIABLE=<code>echo "${line}" | cut -c "$(($Spos+1))-$(($Epos-1))"</code>
      CONTENT=<code>eval echo ${VARIABLE}</code>
      SUFFIX=<code>echo "${line:$Epos}"</code>
      line="\)</span>{PREFIX}<span class="math">\({CONTENT}\)</span>{SUFFIX}"
    else
      break
    fi
  done
  echo "${line}" &gt;&gt; $OUTPUT_FILE
}</p>
<p>echo -n "Path to params file (.json): "
read PARAMS_FILE</p>
<h1>Convert each element of json to a variable.</h1>
<p>KEYS=<code>jq -r 'keys[]' $PARAMS_FILE</code>
for key in $KEYS; do
  eval $key=<code>jq -r .${key} $PARAMS_FILE</code>
done</p>
<p>OUTPUT_FILE="${ID}/batch_u_qsub.sh"
if [ ! -d ${ID} ]; then
  mkdir ${ID}
fi</p>
<p>echo -n "Path to templates file (.sh): "
read TEMPLATES_FILE</p>
<p>PRE_IFS=<span class="math">\(IFS
IFS=\)</span>'\n'
for line in <code>cat ${TEMPLATES_FILE}</code>
do
  AssignParams <span class="math">\({line}
done
IFS=\)</span>PRE_IFS
```</p>
<h3><a href="theme/img/3A_Basic_Laboratory/RNAseq/templates.sh"><code>templates.sh</code></a> <a href="theme/img/3A_Basic_Laboratory/RNAseq/templates.sh" download>[download]</a></h3>
<p>```sh</p>
<h1>!/bin/sh</h1>
<h1>PBS -q [${n_node}]</h1>
<h1>PBS -l select=[<span class="math">\({n_node}]:ncpus=[\)</span>{n_cpus}]:mpiprocs=[<span class="math">\({n_MPIprocs}]:ompthreads=[\)</span>{n_OMPthreads}]</h1>
<h1>PBS -l walltime=[${walltime}]</h1>
<h1>PBS -W group_list=[${group_list}]</h1>
<p>DB_DIR="db"
RNA_DIR="RNAseq"</p>
<p>function DecompressHandler() {
  : '
  @params <span class="math">\({1} Extension 1
  @params ${2} Extension 2
  @params ${3} File Name (hoge.\)</span>{1}.${2})
  '</p>
<p>n_compressed_fn=${#3}</p>
<p>if [ <span class="math">\({2} = "zip" ]; then
    unzip ${3}
    n_extension=4
  elif [ ${2} = "tar" ]; then
    tar ${3}
    n_extension=4
  elif [ ${2} = "gz" ]; then
    gunzip ${3}
    n_extension=3
  elif [ ${2} = "bz2" ]; then
    bzip2 -d ${3}
    n_extension=4
  elif [ ${2} = "lha" -o ${2} = "lzh"]; then
    lha x ${3}
    n_extension=4
  elif [ ${1} = "tar" ]; then
    tar ${3}
    n_extension=\)</span>((${#2}+5))
  else
    n_extension=0
  fi</p>
<p>de_compressed_fn=<span class="math">\({3:0:\)</span>((<span class="math">\(n_compressed_fn-\)</span>n_extension))}
  mv <span class="math">\(de_compressed_fn "../\)</span>{DB_DIR}/"
  if [ ${n_extension} -ne 0 ]; then
    rm ${3}
  fi</p>
<p>echo "${de_compressed_fn}"
}</p>
<h1>Download, Decompress, Dispose</h1>
<p>function D3() {
  : '
  @params ${1} URL
  '</p>
<p>fn=<code>wget -nv --content-disposition $1 2&gt;&amp;1 |cut -d\" -f2</code></p>
<p>extensions=( <code>echo $fn | tr -s '.' ' '</code>)
  n_extensions=${#extensions[@]}</p>
<p>ext1=<span class="math">\({extensions[\)</span>((<span class="math">\(n_extensions-2))]}
  ext2=\)</span>{extensions[<span class="math">\(((\)</span>n_extensions-1))]}
  DecompressHandler $ext1 $ext2 $fn
}</p>
<h1>=== START ===</h1>
<p>cd <span class="math">\(PBS_O_WORKDIR/\)</span>RNA_DIR}
if [ ! -d $DB_DIR ]; then
  mkdir $DB_DIR
fi
if [ ! -d $RNA_DIR ]; then
  mkdir $RNA_DIR
fi</p>
<h1>1.データの取得</h1>
<p>SRA_FILE=<code>D3 [$RNAseqData]</code>
mv "../<span class="math">\({DB_DIR}/\)</span>SRA_FILE" .
fasterq-dump <span class="math">\(SRA_FILE -v --threads [\)</span>n_OMPthreads] --split-files -O ./</p>
<h1>2.品質チェック</h1>
<p>fastqc -t [<span class="math">\(n_OMPthreads] "\)</span>{SRA_FILE}_1.fastq" "${SRA_FILE}_2.fastq"</p>
<h1>3.マッピング</h1>
<p>REF_GENOME_FILE=<code>D3 [$MappingRefGenome]</code>
time hisat2 -x "../<span class="math">\({DB_DIR}/\)</span>{REF_GENOME_FILE}/genome" -1 "<span class="math">\({SRA_FILE}_1.fastq" -2 "\)</span>{SRA_FILE}_2.fastq" -p [<span class="math">\(n_OMPthreads] -S "hisat_output_[\)</span>{ID}].sam"</p>
<h1>4. IGVに必要なインデックスファイル作成</h1>
<p>samtools view --threads [<span class="math">\(n_OMPthreads] -b "hisat_output_[\)</span>{ID}].sam" -o "hisat_output_[<span class="math">\({ID}].bam"
samtools sort --threads [\)</span>n_OMPthreads] "hisat_output_[<span class="math">\({ID}].bam" -o "hisat_output_[\)</span>{ID}].sorted.bam"</p>
<h1>6. リード数のカウント</h1>
<p>GENE_ANNO_FILE=<code>D3 [$GeneAnnotation]</code>
featureCounts hisat_output_ERR315326.bam -p -t exon -g gene_id -s 0 -T [<span class="math">\(n_OMPthreads] -BC -a "../\)</span>{DB_DIR}/<span class="math">\({GENE_ANNO_FILE}" -o "Counts_BC_[\)</span>{ID}].txt"
featureCounts hisat_output_ERR315326.bam -p -t exon -g gene_id -s 0 -T [<span class="math">\(n_OMPthreads] -MOBC -a "../\)</span>{DB_DIR}/<span class="math">\({GENE_ANNO_FILE}" -o "Counts_MOBC_[\)</span>{ID}].txt"
```</p>
<h3><a href="theme/img/3A_Basic_Laboratory/RNAseq/batch_u_qsub.sh"><code>batch_u_qsub.sh</code></a> <a href="theme/img/3A_Basic_Laboratory/RNAseq/batch_u_qsub.sh" download>[download]</a></h3>
<p>```sh</p>
<h1>!/bin/sh</h1>
<h1>PBS -q 1</h1>
<h1>PBS -l select=1:ncpus=1:mpiprocs=1:ompthreads=8</h1>
<h1>PBS -l walltime=0:30:00</h1>
<h1>PBS -W group_list=gj29</h1>
<p>DB_DIR="db"
RNA_DIR="RNAseq"
function DecompressHandler() {
  : '
  @params <span class="math">\({1} Extension 1
  @params ${2} Extension 2
  @params ${3} File Name (hoge.\)</span>{1}.<span class="math">\({2})
  '
  n_compressed_fn=\)</span>{#3}
  if [ <span class="math">\({2} = "zip" ]; then
    unzip ${3}
    n_extension=4
  elif [ ${2} = "tar" ]; then
    tar ${3}
    n_extension=4
  elif [ ${2} = "gz" ]; then
    gunzip ${3}
    n_extension=3
  elif [ ${2} = "bz2" ]; then
    bzip2 -d ${3}
    n_extension=4
  elif [ ${2} = "lha" -o ${2} = "lzh"]; then
    lha x ${3}
    n_extension=4
  elif [ ${1} = "tar" ]; then
    tar ${3}
    n_extension=\)</span>((<span class="math">\({#2}+5))
  else
    n_extension=0
  fi
  de_compressed_fn=\)</span>{3:0:<span class="math">\(((\)</span>n_compressed_fn-<span class="math">\(n_extension))}
  mv $de_compressed_fn "../\)</span>{DB_DIR}/"
  if [ <span class="math">\({n_extension} -ne 0 ]; then
    rm ${3}
  fi
  echo "\)</span>{de_compressed_fn}"
}</p>
<h1>Download, Decompress, Dispose</h1>
<p>function D3() {
  : '
  @params <span class="math">\({1} URL
  '
  fn=<code>wget -nv --content-disposition $1 2&gt;&amp;1 |cut -d\" -f2</code>
  extensions=( <code>echo $fn | tr -s '.' ' '</code>)
  n_extensions=\)</span>{#extensions[@]}
  ext1=<span class="math">\({extensions-2}
  ext2=\)</span>{extensions-1}
  DecompressHandler $ext1 $ext2 $fn
}</p>
<h1>=== START ===</h1>
<p>cd <span class="math">\(PBS_O_WORKDIR/\)</span>RNA_DIR}
if [ ! -d $DB_DIR ]; then
  mkdir $DB_DIR
fi
if [ ! -d $RNA_DIR ]; then
  mkdir $RNA_DIR
fi</p>
<h1>1.データの取得</h1>
<p>SRA_FILE=<code>D3 ftp://ftp.ddbj.nig.ac.jp/ddbj_database/dra/sralite/ByExp/litesra/ERX/ERX288/ERX288491/ERR315326/ERR315326.sra</code>
mv "../<span class="math">\({DB_DIR}/\)</span>SRA_FILE" .
fasterq-dump $SRA_FILE -v --threads 8 --split-files -O ./</p>
<h1>2.品質チェック</h1>
<p>fastqc -t 8 "<span class="math">\({SRA_FILE}_1.fastq" "\)</span>{SRA_FILE}_2.fastq"</p>
<h1>3.マッピング</h1>
<p>REF_GENOME_FILE=<code>D3 https://cloud.biohpc.swmed.edu/index.php/s/grch38/download</code>
time hisat2 -x "../<span class="math">\({DB_DIR}/\)</span>{REF_GENOME_FILE}/genome" -1 "<span class="math">\({SRA_FILE}_1.fastq" -2 "\)</span>{SRA_FILE}_2.fastq" -p 8 -S "hisat_output_ERR315326.sam"</p>
<h1>4. IGVに必要なインデックスファイル作成</h1>
<p>samtools view --threads 8 -b "hisat_output_ERR315326.sam" -o "hisat_output_ERR315326.bam"
samtools sort --threads 8 "hisat_output_ERR315326.bam" -o "hisat_output_ERR315326.sorted.bam"</p>
<h1>6. リード数のカウント</h1>
<p>GENE_ANNO_FILE=<code>D3 ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.annotation.gtf.gz</code>
featureCounts hisat_output_ERR315326.bam -p -t exon -g gene_id -s 0 -T 8 -BC -a "../<span class="math">\({DB_DIR}/\)</span>{GENE_ANNO_FILE}" -o "Counts_BC_ERR315326.txt"
featureCounts hisat_output_ERR315326.bam -p -t exon -g gene_id -s 0 -T 8 -MOBC -a "../<span class="math">\({DB_DIR}/\)</span>{GENE_ANNO_FILE}" -o "Counts_MOBC_ERR315326.txt"
```</p>
<script type="text/javascript">if (!document.getElementById('mathjaxscript_pelican_#%@#$@#')) {
    var align = "center",
        indent = "0em",
        linebreak = "false";

    if (false) {
        align = (screen.width < 768) ? "left" : align;
        indent = (screen.width < 768) ? "0em" : indent;
        linebreak = (screen.width < 768) ? 'true' : linebreak;
    }

    var mathjaxscript = document.createElement('script');
    mathjaxscript.id = 'mathjaxscript_pelican_#%@#$@#';
    mathjaxscript.type = 'text/javascript';
    mathjaxscript.src = 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.3/latest.js?config=TeX-AMS-MML_HTMLorMML';

    var configscript = document.createElement('script');
    configscript.type = 'text/x-mathjax-config';
    configscript[(window.opera ? "innerHTML" : "text")] =
        "MathJax.Hub.Config({" +
        "    config: ['MMLorHTML.js']," +
        "    TeX: { extensions: ['AMSmath.js','AMSsymbols.js','noErrors.js','noUndefined.js'], equationNumbers: { autoNumber: 'none' } }," +
        "    jax: ['input/TeX','input/MathML','output/HTML-CSS']," +
        "    extensions: ['tex2jax.js','mml2jax.js','MathMenu.js','MathZoom.js']," +
        "    displayAlign: '"+ align +"'," +
        "    displayIndent: '"+ indent +"'," +
        "    showMathMenu: true," +
        "    messageStyle: 'normal'," +
        "    tex2jax: { " +
        "        inlineMath: [ ['\\\\(','\\\\)'] ], " +
        "        displayMath: [ ['$$','$$'] ]," +
        "        processEscapes: true," +
        "        preview: 'TeX'," +
        "    }, " +
        "    'HTML-CSS': { " +
        "        fonts: [['STIX', 'TeX']]," +
        "        styles: { '.MathJax_Display, .MathJax .mo, .MathJax .mi, .MathJax .mn': {color: 'inherit ! important'} }," +
        "        linebreaks: { automatic: "+ linebreak +", width: '90% container' }," +
        "    }, " +
        "}); " +
        "if ('default' !== 'default') {" +
            "MathJax.Hub.Register.StartupHook('HTML-CSS Jax Ready',function () {" +
                "var VARIANT = MathJax.OutputJax['HTML-CSS'].FONTDATA.VARIANT;" +
                "VARIANT['normal'].fonts.unshift('MathJax_default');" +
                "VARIANT['bold'].fonts.unshift('MathJax_default-bold');" +
                "VARIANT['italic'].fonts.unshift('MathJax_default-italic');" +
                "VARIANT['-tex-mathit'].fonts.unshift('MathJax_default-italic');" +
            "});" +
            "MathJax.Hub.Register.StartupHook('SVG Jax Ready',function () {" +
                "var VARIANT = MathJax.OutputJax.SVG.FONTDATA.VARIANT;" +
                "VARIANT['normal'].fonts.unshift('MathJax_default');" +
                "VARIANT['bold'].fonts.unshift('MathJax_default-bold');" +
                "VARIANT['italic'].fonts.unshift('MathJax_default-italic');" +
                "VARIANT['-tex-mathit'].fonts.unshift('MathJax_default-italic');" +
            "});" +
        "}";

    (document.body || document.getElementsByTagName('head')[0]).appendChild(configscript);
    (document.body || document.getElementsByTagName('head')[0]).appendChild(mathjaxscript);
}
</script>
  <div>
</div>

  
  
  <hr/>

    <aside class="article_meta">
      <nav>
        <ul class="articles-timeline">
            <li class="previous-article">« <a href="https://iwasakishuto.github.io/University/3A/情報基礎実験(浅井)-11.html" title="Previous: RNA-seqデータ解析（対話モード版）">RNA-seqデータ解析（対話モード版）</a></li>
            <li class="next-article"><a href="https://iwasakishuto.github.io/University/3A/分子生命科学Ⅲ-4.html" title="Next: 分子生命科学Ⅲ 第4回">分子生命科学Ⅲ 第4回</a> »</li>
        </ul>
      </nav>
    </aside>
  </div>
    <section class="article_meta">
      <div class="span2" style="float:right;font-size:0.9em; text-align: center;">
        <a class="btn-square-pop" onclick="switchVisibility('_article_meta_')">hidden</a>
      </div>

      <div class="span2" id="_article_meta_" style="float:right;font-size:0.9em;">
        <h5>Table of Contents</h5>
        <div id="toc"></div>

          <h5>Published</h5>
          <time itemprop="dateCreated" datetime="2019-10-17T00:00:00+09:00">10 17, 2019</time>

<h5>Last Updated</h5>
<time datetime="2019-10-17T00:00:00+09:00">10 17, 2019</time>

          <h5>Category</h5>
          <a class="category-link" href="https://iwasakishuto.github.io/University/3A/categories.html#qing-bao-ji-chu-shi-yan-qian-jing-ref">情報基礎実験(浅井)</a>
          <h5>Tags</h5>
          <ul class="list-of-tags tags-in-article">
              <li><a href="https://iwasakishuto.github.io/University/3A/tags#3a-ref">3A
                  <span>127</span>
</a></li>
              <li><a href="https://iwasakishuto.github.io/University/3A/tags#qing-bao-ji-chu-shi-yan-qian-jing-ref">情報基礎実験(浅井)
                  <span>13</span>
</a></li>
          </ul>
<h5>Contact</h5>
    <a href="https://twitter.com/cabernet_rock" title="My twitter Profile" class="sidebar-social-links" target="_blank">
    <i class="fab fa-twitter sidebar-social-links"></i></a>
    <a href="https://github.com/iwasakishuto" title="My github Profile" class="sidebar-social-links" target="_blank">
    <i class="fab fa-github sidebar-social-links"></i></a>
    <a href="https://www.facebook.com/iwasakishuto" title="My facebook Profile" class="sidebar-social-links" target="_blank">
    <i class="fab fa-facebook sidebar-social-links"></i></a>
      </div>
    </section>
  </div>
</article>
                </div>
                <div class="span1"></div>
            </div>
        </div>
        <div id="push"></div>
    </div>
<footer class="article_meta">
  <div id="footer">
    <ul class="footer-content">
        <li class="elegant-subtitle"><span class="site-name">3A</span> - Shuto's Notes</li>
        <li class="elegant-license">MIT</li>
      <li class="elegant-power">Powered by <a href="http://getpelican.com/" title="Pelican Home Page">Pelican</a>. Theme: <a href="https://github.com/Pelican-Elegant/elegant/" title="Theme Elegant Home Page">Elegant</a></li>
    </ul>
  </div>
</footer>            <script src="https://code.jquery.com/jquery.min.js"></script>
        <script src="https://netdna.bootstrapcdn.com/twitter-bootstrap/2.3.2/js/bootstrap.min.js"></script>
        <script>
            function validateForm(query)
            {
                return (query.length > 0);
            }
        </script>
        <script src="https://iwasakishuto.github.io/js/custom.js"></script>
        <script src="https://iwasakishuto.github.io/js/smooth-scroll.polyfills.min.js"></script>
        <script src="https://iwasakishuto.github.io/js/toc.min.js"></script>
        <script>
            jQuery('#toc').toc({
                'container': 'div.article-content',
                'selectors': 'h2, h3, h4',
            });
        </script>

    
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/css/jupyter.css">
    </body>
    <!-- Theme: Elegant built for Pelican
    License : MIT -->
</html>