<!DOCTYPE html>
<html lang="ja">
    <head>
        <link rel="shortcut icon" href="https://iwasakishuto.github.io/images/apple-touch-icon/University.png" />

        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <meta name="author" content="Shuto" />

        <meta property="og:type" content="article" />
        <meta name="twitter:card" content="summary">

<meta name="keywords" content="3A, 情報基礎実験(浅井), 情報基礎実験(浅井), " />
<meta property="og:image" content="https://iwasakishuto.github.io/images/FacebookImage/University.png"/>

<meta property="og:title" content="RNA-seqデータ解析（バッチスクリプト版） "/>
<meta property="og:url" content="https://iwasakishuto.github.io/University/3A/情報基礎実験(浅井)-12.html" />
<meta property="og:description" content="" />
<meta property="og:site_name" content="3A" />
<meta property="og:article:author" content="Shuto" />
<meta property="og:article:published_time" content="2019-10-17T00:00:00+09:00" />
<meta property="og:article:modified_time" content="2019-10-17T00:00:00+09:00" />
<meta name="twitter:title" content="RNA-seqデータ解析（バッチスクリプト版） ">
<meta name="twitter:description" content="">

        <title>RNA-seqデータ解析（バッチスクリプト版）  · 3A
</title>
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/css/bootstrap-combined.v2.3.2.min.css">
        <link href="//netdna.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.css" rel="stylesheet">
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/University/3A/theme/css/pygments.css" media="screen">
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/University/3A/theme/tipuesearch/tipuesearch.css" media="screen">
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/University/3A/theme/css/elegant.css">
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/University/3A/theme/css/admonition.css" media="screen">
        <!---->
        <link rel="apple-touch-icon" sizes="152x152" href="https://iwasakishuto.github.io/images/apple-touch-icon/University.png" type="image/png" />
<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
     (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
     m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
     })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
    ga('create', 'UA-164333757-1', 'auto');
    ga('send', 'pageview');
</script>



        <!-- Use fontawesome Icon -->
        <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.0/css/all.css" integrity="sha384-lZN37f5QGtY3VHgisS14W3ExzMWZxybE1SJSEsQp9S+oqd12jhcu+A56Ebc1zFSJ" crossorigin="anonymous">
        <!-- Syntax highlight -->
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/8.6/styles/github.min.css">
        <!-- Custom CSS -->
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/css/custom.css">
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/8.6/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script>
        <!-- LaTex -->
        <!-- Github env -->
        <!--<script type="text/javascript" async src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML"></script>-->
        <script async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS_CHTML"></script>
        <script type="text/x-mathjax-config">
        	MathJax.Hub.Config({
        		tex2jax: {
        			inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        			displayMath: [ ['$$','$$'], ["\\[","\\]"] ]
        		}
        	});
        </script>
        <!-- Mermaid -->
        <script src="https://unpkg.com/mermaid/dist/mermaid.min.js" charset="UTF-8"></script>
        <script>
          mermaid.initialize({
            startOnLoad:true
          });
        </script>
        <script src="https://iwasakishuto.github.io/js/smooth-scroll.polyfills.min.js"></script>
    </head>
    <body>
        
        <div id="content-sans-footer">
        <div class="navbar navbar-static-top">
            <div class="navbar-inner">
                <div class="container-fluid">
                    <a class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </a>
                    <a class="brand" href="https://iwasakishuto.github.io/University/3A"><span class=site-name style="color: #80273F;"><i class="fa fa-book"></i> 3A</span></a>
                    <!--
                    <a class="brand" href="https://iwasakishuto.github.io/University/3A/"><span class=site-name style="margin-left:auto; margin-right:auto;"><i class="fas fa-book-reader"></i>3A</span></a>
                    -->
                    <div class="nav-collapse collapse">
                        <ul class="nav pull-right top-menu">
                            <li ><a href="https://iwasakishuto.github.io/"><i class="fas fa-home"></i>Portfolio Top</a></li>
                            <li ><a href="https://iwasakishuto.github.io/University/3A/categories"><i class="fa fa-list-alt"></i> Categories</a></li>
                            <li ><a href="https://iwasakishuto.github.io/University/3A/tags"><i class="fa fa-tags"></i> Tags</a></li>
                            <li ><a href="https://iwasakishuto.github.io/University/3A/archives"><i class="fa fa-folder-open"></i> Archives</a></li>
                            <li><form class="navbar-search" action="https://iwasakishuto.github.io/University/3A/search.html" onsubmit="return validateForm(this.elements['q'].value);"> <input type="text" class="search-query" placeholder="Search" name="q" id="tipue_search_input"></form></li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
        <div class="container-fluid">
            <div class="row-fluid">
                <div class="span1"></div>
                <div class="span10">
<article itemscope>
<div class="row-fluid">
  <header class="page-header span10 offset2">
    <h1><a href="https://iwasakishuto.github.io/University/3A/情報基礎実験(浅井)-12.html"> RNA-seqデータ解析（バッチスクリプト版）  </a></h1>
  </header>
</div>

<div class="row-fluid">
    <div class="span8 offset2 article-content">

  
  <p>今回は、<a href="./情報基礎実験(浅井)-11.html">RNA-seqデータ解析（対話モード版）</a>で行なった処理を毎回対話モードで行うのは面倒なので、バッチスクリプトを作成して一括で行います。</p>
<p>以下のパラメタファイル（<a href="theme/img/3A_Basic_Laboratory/RNAseq/params.json" download><code>params.json</code></a>）を読み込んで各種操作を行います。</p>
<pre class="highlight"><code class="language-json">{
  &quot;qsub_q&quot;: &quot;u-debug&quot;,
  &quot;n_node&quot;: 1,
  &quot;n_cpus&quot;: 1,
  &quot;n_MPIprocs&quot;: 1,
  &quot;n_OMPthreads&quot;: 8,
  &quot;walltime&quot;: &quot;0:30:00&quot;,
  &quot;group_list&quot;: &quot;gj29&quot;,

  &quot;ID&quot;: &quot;ERR315326&quot;,

  &quot;RNAseqData&quot;: &quot;ftp://ftp.ddbj.nig.ac.jp/ddbj_database/dra/sralite/ByExp/litesra/ERX/ERX288/ERX288491/ERR315326/ERR315326.sra&quot;,
  &quot;MappingRefGenome&quot;: &quot;https://cloud.biohpc.swmed.edu/index.php/s/grch38/download&quot;,
  &quot;GeneAnnotation&quot;: &quot;ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.annotation.gtf.gz&quot;
}</code></pre>


<p>各変数は、 <a href="https://stedolan.github.io/jq/download/"><code>jq</code>コマンド</a>を使って以下のように読み込みます。</p>
<pre class="highlight"><code class="language-sh">[変数名]=$(cat [パラメタファイルへのパス] | jq -r .[key名])
[変数名]=`jq -r .[key名] [パラメタファイルへのパス]`</code></pre>


<h2>成果物</h2>
<p><code>make_batch_u_qsub.sh</code> という以下のシェルスクリプトを叩くことで、<code>params.json</code> の内容を踏まえたバッチスクリプト <code>batch_u_qsub.sh</code> を作成します。</p>
<h3><a href="theme/img/3A_Basic_Laboratory/RNAseq/make_batch_u_qsub.sh"><code>make_batch_u_qsub.sh</code></a> <a href="" download>[download]</a></h3>
<pre class="highlight"><code class="language-sh">#!/bin/bash

Ssymbol=&quot;[&quot;
Esymbol=&quot;]&quot;

function AssignParams () {
  &lt;&lt; COMMENTOUT
  @params params.json     {&quot;qsub_q&quot;: &quot;u-debug&quot;}
  @params templates.sh    PBS -q [${n_node}]
  @return batch_u_qsub.sh PBS -q u-debug
  COMMENTOUT

  line=&quot;${1}&quot;
  while true
  do
    Spos=`echo &quot;${line}&quot; | awk -v pattern=${Ssymbol} '{print index($0, pattern)}'`
    Epos=`echo &quot;${line}&quot; | awk -v pattern=${Esymbol} '{print index($0, pattern)}'`
    if [ ${Spos} -ne 0 -a ${Epos} -ne 0 -a ${Spos} -lt ${Epos} ]; then
      PREFIX=`echo &quot;${line:0:$(($Spos-1))}&quot;`
      VARIABLE=`echo &quot;${line}&quot; | cut -c &quot;$(($Spos+1))-$(($Epos-1))&quot;`
      CONTENT=`eval echo ${VARIABLE}`
      SUFFIX=`echo &quot;${line:$Epos}&quot;`
      line=&quot;${PREFIX}${CONTENT}${SUFFIX}&quot;
    else
      break
    fi
  done
  echo &quot;${line}&quot; &gt;&gt; $OUTPUT_FILE
}

echo -n &quot;Path to params file (.json): &quot;
read PARAMS_FILE
# Convert each element of json to a variable.
KEYS=`jq -r 'keys[]' $PARAMS_FILE`
for key in $KEYS; do
  eval $key=`jq -r .${key} $PARAMS_FILE`
done

OUTPUT_FILE=&quot;${ID}/batch_u_qsub.sh&quot;
if [ ! -d ${ID} ]; then
  mkdir ${ID}
fi

echo -n &quot;Path to templates file (.sh): &quot;
read TEMPLATES_FILE

PRE_IFS=$IFS
IFS=$'\n'
for line in `cat ${TEMPLATES_FILE}`
do
  AssignParams ${line}
done
IFS=$PRE_IFS</code></pre>


<h3><a href="theme/img/3A_Basic_Laboratory/RNAseq/templates.sh"><code>templates.sh</code></a> <a href="theme/img/3A_Basic_Laboratory/RNAseq/templates.sh" download>[download]</a></h3>
<pre class="highlight"><code class="language-sh">#!/bin/sh
#PBS -q [${n_node}]
#PBS -l select=[${n_node}]:ncpus=[${n_cpus}]:mpiprocs=[${n_MPIprocs}]:ompthreads=[${n_OMPthreads}]
#PBS -l walltime=[${walltime}]
#PBS -W group_list=[${group_list}]

DB_DIR=&quot;db&quot;
RNA_DIR=&quot;RNAseq&quot;

function DecompressHandler() {
  : '
  @params ${1} Extension 1
  @params ${2} Extension 2
  @params ${3} File Name (hoge.${1}.${2})
  '

  n_compressed_fn=${#3}

  if [ ${2} = &quot;zip&quot; ]; then
    unzip ${3}
    n_extension=4
  elif [ ${2} = &quot;tar&quot; ]; then
    tar ${3}
    n_extension=4
  elif [ ${2} = &quot;gz&quot; ]; then
    gunzip ${3}
    n_extension=3
  elif [ ${2} = &quot;bz2&quot; ]; then
    bzip2 -d ${3}
    n_extension=4
  elif [ ${2} = &quot;lha&quot; -o ${2} = &quot;lzh&quot;]; then
    lha x ${3}
    n_extension=4
  elif [ ${1} = &quot;tar&quot; ]; then
    tar ${3}
    n_extension=$((${#2}+5))
  else
    n_extension=0
  fi

  de_compressed_fn=${3:0:$(($n_compressed_fn-$n_extension))}
  mv $de_compressed_fn &quot;../${DB_DIR}/&quot;
  if [ ${n_extension} -ne 0 ]; then
    rm ${3}
  fi

  echo &quot;${de_compressed_fn}&quot;
}

# Download, Decompress, Dispose
function D3() {
  : '
  @params ${1} URL
  '

  fn=`wget -nv --content-disposition $1 2&gt;&amp;1 |cut -d\&quot; -f2`

  extensions=( `echo $fn | tr -s '.' ' '`)
  n_extensions=${#extensions[@]}

  ext1=${extensions[$(($n_extensions-2))]}
  ext2=${extensions[$(($n_extensions-1))]}
  DecompressHandler $ext1 $ext2 $fn
}

#=== START ===
cd $PBS_O_WORKDIR/$RNA_DIR}
if [ ! -d $DB_DIR ]; then
  mkdir $DB_DIR
fi
if [ ! -d $RNA_DIR ]; then
  mkdir $RNA_DIR
fi

# 1.データの取得
SRA_FILE=`D3 [$RNAseqData]`
mv &quot;../${DB_DIR}/$SRA_FILE&quot; .
fasterq-dump $SRA_FILE -v --threads [$n_OMPthreads] --split-files -O ./
# 2.品質チェック
fastqc -t [$n_OMPthreads] &quot;${SRA_FILE}_1.fastq&quot; &quot;${SRA_FILE}_2.fastq&quot;
# 3.マッピング
REF_GENOME_FILE=`D3 [$MappingRefGenome]`
time hisat2 -x &quot;../${DB_DIR}/${REF_GENOME_FILE}/genome&quot; -1 &quot;${SRA_FILE}_1.fastq&quot; -2 &quot;${SRA_FILE}_2.fastq&quot; -p [$n_OMPthreads] -S &quot;hisat_output_[${ID}].sam&quot;
# 4. IGVに必要なインデックスファイル作成
samtools view --threads [$n_OMPthreads] -b &quot;hisat_output_[${ID}].sam&quot; -o &quot;hisat_output_[${ID}].bam&quot;
samtools sort --threads [$n_OMPthreads] &quot;hisat_output_[${ID}].bam&quot; -o &quot;hisat_output_[${ID}].sorted.bam&quot;
# 6. リード数のカウント
GENE_ANNO_FILE=`D3 [$GeneAnnotation]`
featureCounts hisat_output_ERR315326.bam -p -t exon -g gene_id -s 0 -T [$n_OMPthreads] -BC -a &quot;../${DB_DIR}/${GENE_ANNO_FILE}&quot; -o &quot;Counts_BC_[${ID}].txt&quot;
featureCounts hisat_output_ERR315326.bam -p -t exon -g gene_id -s 0 -T [$n_OMPthreads] -MOBC -a &quot;../${DB_DIR}/${GENE_ANNO_FILE}&quot; -o &quot;Counts_MOBC_[${ID}].txt&quot;</code></pre>


<h3><a href="theme/img/3A_Basic_Laboratory/RNAseq/batch_u_qsub.sh"><code>batch_u_qsub.sh</code></a> <a href="theme/img/3A_Basic_Laboratory/RNAseq/batch_u_qsub.sh" download>[download]</a></h3>
<pre class="highlight"><code class="language-sh">#!/bin/sh
#PBS -q 1
#PBS -l select=1:ncpus=1:mpiprocs=1:ompthreads=8
#PBS -l walltime=0:30:00
#PBS -W group_list=gj29
DB_DIR=&quot;db&quot;
RNA_DIR=&quot;RNAseq&quot;
function DecompressHandler() {
  : '
  @params ${1} Extension 1
  @params ${2} Extension 2
  @params ${3} File Name (hoge.${1}.${2})
  '
  n_compressed_fn=${#3}
  if [ ${2} = &quot;zip&quot; ]; then
    unzip ${3}
    n_extension=4
  elif [ ${2} = &quot;tar&quot; ]; then
    tar ${3}
    n_extension=4
  elif [ ${2} = &quot;gz&quot; ]; then
    gunzip ${3}
    n_extension=3
  elif [ ${2} = &quot;bz2&quot; ]; then
    bzip2 -d ${3}
    n_extension=4
  elif [ ${2} = &quot;lha&quot; -o ${2} = &quot;lzh&quot;]; then
    lha x ${3}
    n_extension=4
  elif [ ${1} = &quot;tar&quot; ]; then
    tar ${3}
    n_extension=$((${#2}+5))
  else
    n_extension=0
  fi
  de_compressed_fn=${3:0:$(($n_compressed_fn-$n_extension))}
  mv $de_compressed_fn &quot;../${DB_DIR}/&quot;
  if [ ${n_extension} -ne 0 ]; then
    rm ${3}
  fi
  echo &quot;${de_compressed_fn}&quot;
}
# Download, Decompress, Dispose
function D3() {
  : '
  @params ${1} URL
  '
  fn=`wget -nv --content-disposition $1 2&gt;&amp;1 |cut -d\&quot; -f2`
  extensions=( `echo $fn | tr -s '.' ' '`)
  n_extensions=${#extensions[@]}
  ext1=${extensions-2}
  ext2=${extensions-1}
  DecompressHandler $ext1 $ext2 $fn
}
#=== START ===
cd $PBS_O_WORKDIR/$RNA_DIR}
if [ ! -d $DB_DIR ]; then
  mkdir $DB_DIR
fi
if [ ! -d $RNA_DIR ]; then
  mkdir $RNA_DIR
fi
# 1.データの取得
SRA_FILE=`D3 ftp://ftp.ddbj.nig.ac.jp/ddbj_database/dra/sralite/ByExp/litesra/ERX/ERX288/ERX288491/ERR315326/ERR315326.sra`
mv &quot;../${DB_DIR}/$SRA_FILE&quot; .
fasterq-dump $SRA_FILE -v --threads 8 --split-files -O ./
# 2.品質チェック
fastqc -t 8 &quot;${SRA_FILE}_1.fastq&quot; &quot;${SRA_FILE}_2.fastq&quot;
# 3.マッピング
REF_GENOME_FILE=`D3 https://cloud.biohpc.swmed.edu/index.php/s/grch38/download`
time hisat2 -x &quot;../${DB_DIR}/${REF_GENOME_FILE}/genome&quot; -1 &quot;${SRA_FILE}_1.fastq&quot; -2 &quot;${SRA_FILE}_2.fastq&quot; -p 8 -S &quot;hisat_output_ERR315326.sam&quot;
# 4. IGVに必要なインデックスファイル作成
samtools view --threads 8 -b &quot;hisat_output_ERR315326.sam&quot; -o &quot;hisat_output_ERR315326.bam&quot;
samtools sort --threads 8 &quot;hisat_output_ERR315326.bam&quot; -o &quot;hisat_output_ERR315326.sorted.bam&quot;
# 6. リード数のカウント
GENE_ANNO_FILE=`D3 ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.annotation.gtf.gz`
featureCounts hisat_output_ERR315326.bam -p -t exon -g gene_id -s 0 -T 8 -BC -a &quot;../${DB_DIR}/${GENE_ANNO_FILE}&quot; -o &quot;Counts_BC_ERR315326.txt&quot;
featureCounts hisat_output_ERR315326.bam -p -t exon -g gene_id -s 0 -T 8 -MOBC -a &quot;../${DB_DIR}/${GENE_ANNO_FILE}&quot; -o &quot;Counts_MOBC_ERR315326.txt&quot;</code></pre>
  <div>
</div>

  
  
  <hr/>

    <aside class="article_meta">
      <nav>
        <ul class="articles-timeline">
            <li class="previous-article">« <a href="https://iwasakishuto.github.io/University/3A/情報基礎実験(浅井)-11.html" title="Previous: RNA-seqデータ解析（対話モード版）">RNA-seqデータ解析（対話モード版）</a></li>
            <li class="next-article"><a href="https://iwasakishuto.github.io/University/3A/分子生命科学Ⅲ-4.html" title="Next: 分子生命科学Ⅲ 第4回">分子生命科学Ⅲ 第4回</a> »</li>
        </ul>
      </nav>
    </aside>
  </div>
    <section class="article_meta">
      <div class="span2" style="float:right;font-size:0.9em; text-align: center;">
        <a class="btn-square-pop" onclick="switchVisibility('_article_meta_')">hidden</a>
      </div>

      <div class="span2" id="_article_meta_" style="float:right;font-size:0.9em;">
        <h5>Table of Contents</h5>
        <div id="toc"></div>

          <h5>Published</h5>
          <time itemprop="dateCreated" datetime="2019-10-17T00:00:00+09:00">Oct 17, 2019</time>

<h5>Last Updated</h5>
<time datetime="2019-10-17T00:00:00+09:00">Oct 17, 2019</time>

          <h5>Category</h5>
          <a class="category-link" href="https://iwasakishuto.github.io/University/3A/categories.html#qing-bao-ji-chu-shi-yan-qian-jing-ref">情報基礎実験(浅井)</a>
          <h5>Tags</h5>
          <ul class="list-of-tags tags-in-article">
              <li><a href="https://iwasakishuto.github.io/University/3A/tags#3a-ref">3A
                  <span>127</span>
</a></li>
              <li><a href="https://iwasakishuto.github.io/University/3A/tags#qing-bao-ji-chu-shi-yan-qian-jing-ref">情報基礎実験(浅井)
                  <span>13</span>
</a></li>
          </ul>
<h5>Contact</h5>
    <a href="https://twitter.com/cabernet_rock" title="My twitter Profile" class="sidebar-social-links" target="_blank">
    <i class="fab fa-twitter sidebar-social-links"></i></a>
    <a href="https://github.com/iwasakishuto" title="My github Profile" class="sidebar-social-links" target="_blank">
    <i class="fab fa-github sidebar-social-links"></i></a>
<h5>Other contents</h5>
<ul class="other-contents"></ul>
    <li><a href="https://iwasakishuto.github.io" target="_blank">Home</a></li>
    <li><a href="https://iwasakishuto.github.io/Blog/index.html" target="_blank">Blog</a></li>
    <li><a href="https://iwasakishuto.github.io/Front-End/index.html" target="_blank">Front-End</a></li>
    <li><a href="https://iwasakishuto.github.io/Kerasy/index.html" target="_blank">Kerasy</a></li>
    <li><a href="https://iwasakishuto.github.io/Python-Charmers/index.html" target="_blank">Python-Charmers</a></li>
    <li><a href="https://iwasakishuto.github.io/Translation-Gummy/index.html" target="_blank">Translation-Gummy</a></li>
</ul>

      </div>
    </section>
  </div>
</article>
                </div>
                <div class="span1"></div>
            </div>
        </div>
        <div id="push"></div>
    </div>
<footer class="article_meta">
  <div id="footer">
    <ul class="footer-content">
        <li class="elegant-subtitle"><span class="site-name">3A</span> - Shuto's Notes</li>
        <li class="elegant-license">MIT</li>
      <li class="elegant-power">Powered by <a href="http://getpelican.com/" title="Pelican Home Page">Pelican</a>. Theme: <a href="https://github.com/Pelican-Elegant/elegant/" title="Theme Elegant Home Page">Elegant</a></li>
    </ul>
  </div>
</footer>            <script src="https://code.jquery.com/jquery.min.js"></script>
        <script src="https://netdna.bootstrapcdn.com/twitter-bootstrap/2.3.2/js/bootstrap.min.js"></script>
        <script>
            function validateForm(query)
            {
                return (query.length > 0);
            }
        </script>
        <script src="https://iwasakishuto.github.io/js/custom.js"></script>
        <script src="https://iwasakishuto.github.io/js/smooth-scroll.polyfills.min.js"></script>
        <script src="https://iwasakishuto.github.io/js/toc.min.js"></script>
        <script>
            jQuery('#toc').toc({
                'container': 'div.article-content',
                'selectors': 'h2, h3, h4',
            });
        </script>

    
        <link rel="stylesheet" type="text/css" href="https://iwasakishuto.github.io/css/jupyter.css">
    </body>
    <!-- Theme: Elegant built for Pelican
    License : MIT -->
</html>